crates/telio-starcast/src/nat.rs

```diff
@@ -0,0 +1,635 @@
+use pnet_packet::{
+    ip::{IpNextHeaderProtocol, IpNextHeaderProtocols},
+    ipv4::{self, Ipv4Packet, MutableIpv4Packet},
+    ipv6::MutableIpv6Packet,
+    udp::{self, MutableUdpPacket},
+    MutablePacket,
+};
+use std::{
+    hash::{DefaultHasher, Hash, Hasher},
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    time::Duration,
+};
+use telio_utils::{telio_log_warn, LruCache};
+
+trait MutableIpPacket<'a>: Sized + MutablePacket {
```

Comment:
Let's move this trait and the impls for v4 and v6 to make it more general, at least to it's own file in the same crate

Reply:
:+1: